pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -649,21 +649,32 @@ func (m *ManagerImpl) devicesToAllocate(podUID, contName, resource string, requi
 	containerTopologyHint := m.topologyAffinityStore.GetAffinity(podUID, contName)
 	klog.Infof("Topology Affinities for pod %v container %v are: %v", podUID, contName, containerTopologyHint)
 
+	allocated := available.UnsortedList()[:needed]
+	availableTopologyAligned := available
+
 	sockets := make(map[int]bool)
 	socketsArray := containerTopologyHint.SocketMask.GetSockets()
```

Comment:
I think this should be SocketAffinity, not SocketMask

Reply:
Added a condition around this section for TopologyHint being nil